src/librender/mesh.cpp

```diff
@@ -890,8 +890,19 @@ MTS_VARIANT void Mesh<Float, Spectrum>::parameters_changed(const std::vector<std
     }
 }
 
+MTS_VARIANT void Mesh<Float, Spectrum>::set_grad_suspended(bool state) {
+    if constexpr (ek::is_diff_array_v<Float>) {
+        ek::set_grad_suspended(m_vertex_positions, state);
+        ek::set_grad_suspended(m_vertex_normals, state);
+        ek::set_grad_suspended(m_vertex_texcoords, state);
+        for(auto &[name, attribute]: m_mesh_attributes)
+            ek::set_grad_suspended(attribute.buf, state);
+    }
+}
+
 MTS_VARIANT bool Mesh<Float, Spectrum>::parameters_grad_enabled() const {
     if constexpr (ek::is_diff_array_v<Float>) {
+        //TODO: add mesh attributes.
```

Comment:
Not a big fan of adding TODOs (these just pile up eternally) -- let's just fix the issue when first encountered :-).

Reply:
Right. I just wanted to do it in a separate commit :)

Reply:
Fair enough :)